src/main/scala/nl/knaw/dans/api/sword2/SwordID.scala

```diff
@@ -17,30 +17,33 @@ package nl.knaw.dans.api.sword2
 
 import org.swordapp.server.SwordError
 
+import scala.util.{Failure, Success, Try}
+
 object SwordID {
-  def generate: String = {
-    val id: Long = System.currentTimeMillis
+  def generate: Try[String] = synchronized {
```

Comment:
*synchronized* could become a bottle neck here, especially if we sleep for 2 ms in it. It will do for now, though.